### ISSUE TITLE
E117: Unknown function: TSOnBufEnter (vim)

### ISSUE THREAD

**Issue by Terseus** [mhartington/nvim-typescript#93]
# Problems summary

This error appears when opening any `.ts` file with the plugin active, and the plugin doesn't work (no autocompletion for `.ts` files):
```
Error detected while processing BufEnter Auto commands for "*.ts":
E117: Unknown function: TSOnBufEnter
```

## Expected

No error happens.

## Environment Information

 * terminal: sakura 3.3.4
* OS: Archlinux x64
* Shell: fish 2.6.0
* vim/nvim version: vim 8.0 patches 1-1297
* Python info:
```
$ python -c 'from sys import version; import neovim; print("Python version: {0!s}".format(version)); print("Python neovim version: {0!s}".format(neovim.VERSION))'
Python version: 3.6.3 (default, Oct 24 2017, 14:48:20)
[GCC 7.2.0]
Python neovim version: Version(major=0, minor=2, patch=0, prerelease='')
```
* Full `vim --version` output:
```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Nov 14 2017 20:05:55)
Included patches: 1-1297
Compiled by Arch Linux
Huge version with GTK3 GUI.  Features included (+) or not (-):
+acl             +farsi           +mouse_netterm   +tag_binary
+arabic          +file_in_path    +mouse_sgr       +tag_old_static
+autocmd         +find_in_path    -mouse_sysmouse  -tag_any_white
-autoservername  +float           +mouse_urxvt     +tcl/dyn
+balloon_eval    +folding         +mouse_xterm     +termguicolors
+browse          -footer          +multi_byte      +terminal
++builtin_terms  +fork()          +multi_lang      +terminfo
+byte_offset     +gettext         -mzscheme        +termresponse
+channel         -hangul_input    +netbeans_intg   +textobjects
+cindent         +iconv           +num64           +timers
+clientserver    +insert_expand   +packages        +title
+clipboard       +job             +path_extra      +toolbar
+cmdline_compl   +jumplist        +perl/dyn        +user_commands
+cmdline_hist    +keymap          +persistent_undo +vertsplit
+cmdline_info    +lambda          +postscript      +virtualedit
+comments        +langmap         +printer         +visual
+conceal         +libcall         +profile         +visualextra
+cryptv          +linebreak       +python/dyn      +viminfo
+cscope          +lispindent      +python3/dyn     +vreplace
+cursorbind      +listcmds        +quickfix        +wildignore
+cursorshape     +localmap        +reltime         +wildmenu
+dialog_con_gui  +lua/dyn         +rightleft       +windows
+diff            +menu            +ruby/dyn        +writebackup
+digraphs        +mksession       +scrollbind      +X11
+dnd             +modify_fname    +signs           -xfontset
-ebcdic          +mouse           +smartindent     +xim
+emacs_tags      +mouseshape      +startuptime     -xpm
+eval            +mouse_dec       +statusline      +xsmp_interact
+ex_extra        +mouse_gpm       -sun_workshop    +xterm_clipboard
+extra_search    -mouse_jsbterm   +syntax          -xterm_save
   system vimrc file: "/etc/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "/etc/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -D_FORTIFY_SOURCE=2  -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: gcc   -L. -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -fstack-protector -rdynamic -Wl,-export-dynamic -Wl,-E -Wl,-rpath,/usr/lib/perl5/5.26/core_perl/CORE  -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -L/usr/local/lib -Wl,--as-needed -o vim   -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lSM -lICE -lXt -lX11 -lXdmcp -lSM -lICE  -lm -ltinfo -lelf -lnsl    -lacl -lattr -lgpm -ldl   -Wl,-E -Wl,-rpath,/usr/lib/perl5/5.26/core_perl/CORE -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -fstack-protector-strong -L/usr/local/lib  -L/usr/lib/perl5/5.26/core_perl/CORE -lperl -lpthread -lnsl -ldl -lm -lcrypt -lutil -lc   -L/usr/lib -ltclstub8.6 -ldl -lz -lpthread -lieee -lm
```

## Provide a minimal vim rc with less than 50 lines (Required!)

```vim
set nocompatible

call plug#begin('~/.vim/bundle')

Plug 'roxma/nvim-yarp'
Plug 'roxma/vim-hug-neovim-rpc'
Plug 'Shougo/deoplete.nvim'
Plug 'mhartington/nvim-typescript'

call plug#end()

let g:deoplete#enable_at_startup = 1
```

Sorry for using `Plug` in the minimal vimrc example but I don't know how to load plugins manually 😅 

## The reproduce ways from vim starting

* Use `vim -u <example file provided> <TS file>`
* The error appear

## Additional information

I've seen #50 but the workaround provided there works only in neovim and I'm using vim, where the command `:UpdateRemotePlugins` doesn't exists.

**Comment by mhartington** [comment]
Yeah, this is probably because it's Vim and not Neovim. I still need to merge #84 to add Vim support

**Comment by karsai5** [comment]
I'm getting the same problem with neovim. 
```
$ nvim --version

NVIM v0.2.1
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/local/Homebrew/Library/Homebrew/shims/super/clang -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -I/tmp/neovim-20171109-5288-goamgr/neovim-0.2.1/build/config -I/tmp/neovim-20171109-5288-goamgr/neovim-0.2.1/src -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/opt/gettext/include -I/usr/include -I/tmp/neovim-20171109-5288-goamgr/neovim-0.2.1/build/src/nvim/auto -I/tmp/neovim-20171109-5288-goamgr/neovim-0.2.1/build/include
Compiled by brew@HighSierra-2.local

Features: +acl +iconv +jemalloc +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.2.1/share/nvim"

Run :checkhealth for more info
```

**Comment by karsai5** [comment]
Fixed with `:UpdateRemotePlugins`

**Comment by mhartington** [comment]
Closing for now as I need to work on the vim8 portion more in my free time.

**Comment by eshinn** [comment]
@karsai5 - I'm using Vim8 and it doesn't understand `:UpdateRemotePlugins` -- I'm assuming that's an nvim function?

I've installed both: `vim` & `macvim` with brew (tried one then the other), `roxma/vim-hug-neovim-rpc/issues`, `roxma/nvim-yarp`, `shougo/deoplete.nvim`, `pynvim (pip3 install pynvim)`

...it seems like everything works fine when checking each individually - they just don't play along. Anyone have this issue? I'm assuming `:UpdateRemotePlugins` is nvim, which would be connected to neovim (which I installed via `npm i -g`) and should connect via the two shougo plugins.

Halp!
